In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.read_csv("\\Users\\piyus\\Documents\\data assiogh\\Hypothesis-Testing_Costomer-OrderForm-main\\Costomer+OrderForm.csv")

In [3]:
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [5]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [6]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [7]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [8]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [9]:
data = pd.DataFrame(index=['Error Free', 'Defective'],
                    data={'Phillippines':[271,29], 'Indonesia':[267,33], "Malta":[269,31], "India":[280,20]})

In [10]:
data

Phillippines  Indonesia  Malta  India
Error Free           271        267    269    280
Defective             29         33     31     20

In [11]:
from scipy.stats import chi2_contingency

In [12]:
stat, p, dof, expected = stats.chi2_contingency(data)

In [13]:
stat

3.858960685820355

In [14]:
p     #p-Value

0.2771020991233135

since the p-Value (0.277) is greater than alpha (0.05), hence we failed to Reject ho.

